#163860903 Navigation Bar

Georgeygigz opened the pull request:
**What does this PR do?**
* It adds a  Navigation bar

**Description of Task to be completed?**

* Adding a Navigation bar.

**How should this be manually tested?**

* clone this repo `https://github.com/andela/ah-django-unchained-frontend`
* `cd ah-django-unchained-frontend`
* `git checkout ft-navbar-163860903`
* Install the required dependencies `npm install`
* `npm start to run the local server`

**Pivotal Tracker story**
[#163860903](https://www.pivotaltracker.com/story/show/163860903)

![image](https://user-images.githubusercontent.com/24627269/52598515-fd234600-2e66-11e9-9c3e-1e61a6f9a824.png)

Georgeygigz committed:
feat(navbar):Navigation Bar

-Add Navigation bar

[Starts #163860903]

cosmas28 commented:
@Georgeygigz kindly fix the failing tests on travis

cosmas28 commented on src/App.jsx:
```diff
@@ -1,9 +1,15 @@
 const App = () => {
   return (
+    <div>
+    <NavBar />
```
Kindly indent the `<NavBar/>` and `<Routes/>` here. This will make the code clean, consistent and readable.

Georgeygigz replied:
Okay working on it

cosmas28 commented on src/pages/NavBar.jsx:
```diff
@@ -0,0 +1,42 @@
+import React, { Component } from 'react';
```
Move the This component to the `src/components/NavBar/index.jsx` since it is going to be used mostly as a presentational component rather than a container component.

Georgeygigz replied:
Okay working on it

cosmas28 commented on src/routes/routes.jsx:
```diff
@@ -6,12 +6,12 @@ import Login from '../components/Login';
     </BrowserRouter>
   );
 };
-
```
Any reason why you removed this line here?

Georgeygigz replied:
Okay will fix this

cosmas28 commented on src/routes/routes.jsx:
```diff
@@ -6,12 +6,12 @@ import Login from '../components/Login';
 };
-
-export default Routes;
+export default Routes
```
There should a line space at the end of each file.

cosmas28 commented on src/pages/NavBar.jsx:
```diff
@@ -0,0 +1,42 @@
+      isOpen: !this.state.isOpen
+    });
+  }
+  render() {
```
There should be a line space before any method definition.

cosmas28 commented on src/pages/NavBar.jsx:
```diff
@@ -0,0 +1,42 @@
+  constructor(props) {
+    super(props);
+
+    this.navBarToggle = this.navBarToggle.bind(this);
```
Any reason why you bind this toggle method? If you use the arrow function technique, the `this` is bound automatically.

cosmas28 commented on src/pages/NavBar.jsx:
```diff
@@ -0,0 +1,42 @@
+      isOpen: false
+    };
+  }
+  navBarToggle() {
```
Kindly adhere to the method naming conventions. All methods should be in form of a verb, for instance, this method should be like `toggleNavBar`.

cosmas28 replied:
Also use the `Arrow` function technique here to avoid binding the `this` keyword.

Georgeygigz committed:
feat(Add navbar to components):Add navigation bar to components

- Add navigation bar to component folder
- Remove linting errors

[Fixes #163860903]

Georgeygigz committed:
feat(test):Add test for the NavBar

- Test the NavBar component

[Fixes #163860903]

cosmas28 commented:
The head of your commit messages is the same, in such its advisable to have one commit message. So, please stash your commit messages.

cosmas28 commented on src/components/NavBar/NavBar.test.jsx:
```diff
@@ -0,0 +1,21 @@
+import NavBar from './index';
+
+describe('Navigation Bar :', () => {
+    let enzymeWrapper;
```
This variable name is not descriptive enough, I would prefer a name like `wrapper` or `componentWrapper`. What do you think?

cosmas28 commented on src/components/NavBar/NavBar.test.jsx:
```diff
@@ -0,0 +1,21 @@
+import { mount } from 'enzyme';
+import NavBar from './index';
+
+describe('Navigation Bar :', () => {
```
The describe block title is not intuitive enough, what do you think about `The NavBar`? Also, the colon is not necessary.

cosmas28 commented on src/components/NavBar/NavBar.test.jsx:
```diff
@@ -0,0 +1,21 @@
+    });
+
+    it('test if component mounts without crashing', () => {
+        expect(enzymeWrapper).not.toBe(undefined);
```
any reason why you are not using snapshot testing here?

cosmas28 replied:
To fix the `Travis issues` please modify the `script:` in `.travis.yml` from `npm test` to `npm test -- -u`. You should also gitignore the snapshot files.

Georgeygigz replied:
> any reason why you are not using snapshot testing here?

I will work on that too

cosmas28 commented on src/App.jsx:
```diff
@@ -1,9 +1,14 @@
 const App = () => {
   return (
-    <Routes />
+    <div>
```
What do you think about using `<React.Fragment>` instead of empty divs?

cosmas28 commented on src/components/NavBar/index.jsx:
```diff
@@ -0,0 +1,51 @@
+                <NavLink to="signup">SignUp</NavLink>
+              </NavItem>
+              <NavItem>
+                <NavLink to="/login">Login</NavLink>
```
Thess links are not working on browser, kindly create a `Bug` type story to configure the routes to enable us to use `Link` component from `react-router-dom`.

Georgeygigz commented:
> The head of your commit messages is the same, in such its advisable to have one commit message. So, please stash your commit messages.
@cosmas28  I have only have only two commits
